### ISSUE TITLE
`macos-*` images are suddenly lacking `pcre`

### ISSUE THREAD

**Issue by firewave** [actions/runner-images#6634]
### Description

Starting November 23 some builds on `macos-*` runners started to fail since `pcre` is no longer installed. Curiously this did not apply to all the runners then and some builds did still complete.

I assumed this might be a brownout to raise awareness but the builds are consistently failing now. I didn't see any deprecation warnings and could not find any announcements that the images are being changed.

The library is also not mentioned at all in the macos readmes:
https://github.com/actions/runner-images/blob/1f84dd7159610d519a3bdc2beff2352b41a9355f/images/macos/macos-11-Readme.md
https://github.com/actions/runner-images/blob/3a2d6530a3c8c0a7fedcfdce32d5936a3344a17a/images/macos/macos-11-Readme.md

I am aware though that `pcre` has been EOL for quite a while now and should be replaced with `pcre2`.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [X] macOS 11
- [X] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Early failure: https://github.com/danmar/cppcheck/actions/runs/3534619139/jobs/5931705729
```
2022-11-23T18:39:42.7369810Z Image: macos-11
2022-11-23T18:39:42.7370370Z Version: 20221121.1
```

Successful build from the same workflow invocation: https://github.com/danmar/cppcheck/actions/runs/3534619139/jobs/5931705111
```
2022-11-23T18:37:30.2486600Z Image: macos-11
2022-11-23T18:37:30.2487150Z Version: 20221028.1
```

Failure from a recent invocation of some workflow: https://github.com/danmar/cppcheck/actions/runs/3534619139/jobs/5958652770

### Is it regression?

Yes

### Expected behavior

`pcre` is available.

### Actual behavior

`pcre` is not available.

### Repro steps

Try to run the `pcre-config` command on a `macos-*` runner.

**Comment by igorboskovic3** [comment]
Hi @firewave we will check it.

**Comment by igorboskovic3** [comment]
Hi @firewave I can confirm that we have `pcre2` on our images

![image](https://user-images.githubusercontent.com/68074361/204002933-a27c13f4-3d3e-48f9-81f2-e76cb772e9a8.png)

If you want to install `pcre`, you could use
`brew install pcre`

Feel free to reach us if you have more questions.

**Comment by firewave** [comment]
`pcre2` is not the same as `pcre` and they are not compatible and require code changes.

`20221121.1` no longer has `pcre` which causes builds to fail. It was still available in `20221028.1` (that's for `macos-11`).

**Comment by chenrui333** [comment]
> `pcre2` is not the same as `pcre` and they are not compatible and require code changes.
> 
> `20221121.1` no longer has `pcre` which causes builds to fail. It was still available in `20221028.1` (that's for `macos-11`).

Maybe just you need `brew install pcre` step in your workflow run?

**Comment by firewave** [comment]
> Maybe just you need `brew install pcre` step in your workflow run?

Obviously that will fix it (and is what I did).

But as mentioned before the package disappeared without notice from the image. The library was not listed as being provided by the image and is EOL but still...
